lib/rules/no-restricted-exports.js

```diff
@@ -109,7 +112,15 @@ module.exports = {
         function checkExportedName(node) {
             const name = astUtils.getModuleExportName(node);
 
-            if (restrictedNames.has(name)) {
+            let hasRestrictedNamePattern = false;
+
+            if (restrictedNamePattern) {
+                const patternRegex = new RegExp(restrictedNamePattern, "u");
+
+                hasRestrictedNamePattern = name && patternRegex.test(name);
```

Comment:
```suggestion
                hasRestrictedNamePattern = patternRegex.test(name);
```

At this point, `name` is certainly a string, and a falsy empty string name should also be checked against the pattern.

```js
/*eslint no-restricted-exports: ["error", {
    "restrictedNamedExportsPattern": "^$"
}]*/


const foo = 1;

export { foo as "" }; // false negative, it matces the pattern

```